src/control/security/auth/auth_sys.go

```diff
@@ -146,6 +146,16 @@ func sysNameToPrincipalName(name string) string {
 	return name + "@"
 }
 
+func GetMachineName() (string, error) {
```

Comment:
minor: This is a public function, so it should have a docstring.

Reply:
nit: The point I was trying to make with my other comment was that the format of a machine name is an implementation detail which could change, hence the advice to refactor into a common implementation that will ensure compatibility across code that needs to get at a machine name that will be or was stored in a pool handle.

It's perhaps a subtle distinction, but for a helper like this, IMO the docstring shouldn't describe the implementation, but the purpose/intended use of the function. In this case, the intended use is to generate a machine name suitable for storage in a pool handle. If you wanted to go overboard (I don't recommend this), then you would probably add a utility function called `GetShortHostname()` which does as it says on the tin, and then `GetMachineName()` would call that function. That way if the requirements for what gets stored in the pool handle change, you only need to change the implementation of `GetMachineName()` and callers don't know/don't care about the details. A suitably descriptive-but-generalized docstring wouldn't need to be updated, either.

Not asking for a change here, just some food for thought.